internal/controller/controllers/installenv_controller.go

```diff
@@ -140,6 +144,40 @@ func (r *InstallEnvReconciler) updateClusterDiscoveryIgnitionIfNeeded(ctx contex
 	return nil
 }
 
+func (r *InstallEnvReconciler) processNMStateConfig(ctx context.Context, installEnv *adiiov1alpha1.InstallEnv) ([]*models.HostStaticNetworkConfig, error) {
+	var staticNetworkConfig []*models.HostStaticNetworkConfig
+
+	if installEnv.Spec.NMStateConfigLabelSelector.MatchLabels == nil {
+		return staticNetworkConfig, nil
+	}
+	NMStateLabelValue, found := installEnv.Spec.NMStateConfigLabelSelector.MatchLabels[SelectorNMStateConfigNameLabel]
+	if !found {
+		return staticNetworkConfig, nil
+	}
+	nmStateConfigs, err := getNMStateConfigsByLabelValue(ctx, r.Client, installEnv.Namespace, NMStateLabelValue)
+	if err != nil {
+		return staticNetworkConfig, err
+	}
+	for _, nmStateConfig := range nmStateConfigs.Items {
+		r.Log.Debugf("found nmStateConfig: %s for installEnv: %s", nmStateConfig.Name, installEnv.Name)
+		macInterfaceMap := models.MacInterfaceMap{}
```

Comment:
suggesting to use make it will be faster, you know the size by `len(nmStateConfig.Spec.MacInterfaceMap)`

Reply:
Done